src/google/protobuf/repeated_ptr_field.h

```diff
@@ -412,7 +412,7 @@ class PROTOBUF_EXPORT RepeatedPtrFieldBase {
     // our arena pointer, and we can add to array without resizing it (at
     // least one slot that is not allocated).
     void** elems = elements();
-    if (current_size_ < allocated_size()) {
+    if ((current_size_ > 0) && (current_size_ < allocated_size())) {
```

Comment:
How is `current_size_` ending up negative? I think we should understand why that is happening before determining the right fix.

Reply:
Hi Adam,

How is current_size_ ending up negative ? - that's outside the scope of this commit, this commit simply tries to behave better when it does happen.

Reply:
I don't think we should try to fix this without understanding why this happening. Actually the fact that this is a compile-time error makes me think that `current_size_` is not negative but instead there's something else about the way the code is structured that GCC 13 doesn't like.